core/Azure.Mcp.Core/src/Areas/Server/Commands/ServiceStartCommand.cs

```diff
@@ -708,4 +712,21 @@ private static string ResolveTransport(CommandResult commandResult)
     {
         return commandResult.GetValueOrDefault<string>(ServiceOptionDefinitions.Transport.Name) ?? TransportTypes.StdIo;
     }
+
+    private static WebApplication UseHttpsRedirectionIfEnabled(WebApplication app)
+    {
+        // Some hosting environments may not need HTTPS redirection, such as:
+        // - Running behind a reverse proxy that handles TLS termination.
+        // - Local development when not using self-signed development certs.
+        // - The application or server's HTTP stack is not listening for non-HTTPS requests.
+        //
+        // Safe default to enable HTTPS redirection unless explicitly opted-out.
+        string? httpsRedirectionOptOut = Environment.GetEnvironmentVariable("AZURE_MCP_DISABLE_HTTPS_REDIRECTION");
```

Comment:
Discussed this env var name internally. and agreed on `AZURE_MCP_DANGEROUSLY_DISABLE_HTTPS_REDIRECTION`

1. Using the `AZURE_MCP_ ` product prefix makes it clear that this environment variable belongs to azmcp.
2. The `_DANGEROUSLY_` portion is consistent with other place that allow users to override secure defaults (e.g., --dangerously-* flags).